src/libclipper/test/json_util_test.cpp

```diff
@@ -0,0 +1,206 @@
+#include <gtest/gtest.h>
+
+#include <clipper/datatypes.hpp>
+#include <clipper/json_util.hpp>
+
+using namespace clipper_json;
+
+/* Test JSON serialization utilities */
+TEST(JsonUtilTests, TestCorrectJsonValues) {
+  std::string expected_json =
+      "{\"string_val\":\"test_string\",\"double_val\":0.3,\"int_val\":-100}";
+  std::string string_val = "test_string";
+  double double_val = 0.3;
+  int int_val = -100;
+
+  rapidjson::Document d;
+  d.SetObject();
+  add_string(string_val, "string_val", d);
+  add_double(double_val, "double_val", d);
+  add_int(int_val, "int_val", d);
+  std::string output_json = to_json_string(d);
+  EXPECT_EQ(output_json, expected_json); 
+}
+
+TEST(JsonUtilTests, TestCorrectJsonArrays) {
+  std::string expected_json =
+      "{\"string_array\":[\"test1\",\"word2\",\"phrase3\"],"
```

Comment:
Anywhere that you need to use an excessive number of escape characters (for quotes, new lines, etc) you can just use a raw string literal.